File: Go/fizzbuzz.go
package main

import (
	"fmt"
	"strconv"
)

func FizzBuzz(n int) string {
	fizz := (n % 3) == 0
	buzz := (n % 5) == 0
	switch {
	case fizz && buzz:
		return "FizzBuzz"
	case fizz:
		return "Fizz"
	case buzz:
		return "Buzz"
	default:
		return strconv.Itoa(n)
	}
}

func main() {
	for i := 1; i <= 100; i++ {
		fmt.Println(FizzBuzz(i))
	}
}

File: OCaml/fizzbuzz.ml
let fizzbuzz n =
    match ((n mod 3), (n mod 5)) with
        | (0, 0) -> "FizzBuzz"
        | (0, _) -> "Fizz"
        | (_, 0) -> "Buzz"
        | (_, _) -> string_of_int n
